### PR TITLE
Wb tournament owner check

### PR DESCRIPTION
**Pull request by williambertrand**
### Description
- Only show "Add Brackets" and "Start tournament" if the current user is the tournament owner
- Placeholder for submitting a Start tournament request

### CODE REVIEW

File: models.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
+import { User as FBUser } from 'firebase/auth'
+
+export interface User extends FBUser {
```
Comment (probicheaux) [inline models.tsx]: Hmmm, is this a thing? I feel like the User response from the server doesn't extend FBUser. Am I confused about something?

File: screens/TournamentDetails.tsx [hunk 2]
```diff
@@ -43,15 +44,23 @@ export default function TournamentDetailsScreen({
...
+    setLoading(true);
+    await startTournament(tournamentId, user as User);
+    fetchTournamentDetails();
+  }
+
   if (loading || tournament === undefined) {
     return <Spinner />;
   }
 
+  const isOwner = owner?.firebase_id === user?.uid;
```
Comment (probicheaux) [inline screens/TournamentDetails.tsx:64]: noice

**Review by probicheaux — APPROVED** [review]
lgtm assuming you explain to me the one question I have

File: models.tsx (hunk 1 above)
Comment (williambertrand) [reply models.tsx]: Ouch actually yeah this is bad lol. Not sure what I was thinking exactly when I did this. 

I did this as a way to cheat into not having to say whether we're dealing with a User from our API or a User from Firebase down when dealing with the Owner below. But really we should have our own `BrackendUser` interface or something. Lemme fix this up

Comment (williambertrand) [reply models.tsx]: @probicheaux I made a `BracketUser` type to represent users from our API, and then we'll use the Firebase User type when looking at the current user context: https://github.com/probicheaux/ReactBracket/pull/18/commits/b1189b7eaff42739212daab20a8b294e549a9e04

**Pull request merged**